repository_service_tuf_api/targets.py

```diff
@@ -66,7 +66,13 @@ class AddPayload(BaseModel):
     """
 
     targets: List[Targets]
-    add_task_id_to_custom: Optional[bool]
+    add_task_id_to_custom: Optional[bool] = Field(
+        default=False,
+        description="Whether to add the id of the task in custom",
+    )
+    publish_targets: Optional[bool] = Field(
```

Comment:
Hey, just passing by here, a small comment. Shouldn't the type be `bool` (instead of `Optional[bool]`) since we give a default. Same for [line 90](https://github.com/vmware/repository-service-tuf-api/pull/254/files#diff-04e45fdfaa48b6ef3b8855ee0940eaeccf5b9010e5613a74fdaf617b351da35aR90)

Reply:
Making it optional means that the user is not required to give any value to the `publish_targets` attribute.
If no value is given, then pydantic will make sure to populate it.
We do that to escape the case when we have `publish_targets` None and we have missed handling it.

Reply:
You can also make it optional by setting it as `bool = Field(default=True, description="Whether to publish the targets")`, I tested both ways. What I like about the aforementioned typing is that when you use it later on, for example in:
```python
def post(payload: AddPayload) -> Response:
...
...
    if payload.publish_targets is False:
...
``` 
you can easily tell that `payload.publish_targets` is a `bool` (even when not provided by the user, since it has a default) and not `None`.

In any case, I tested in `mypy` with both typing and it doesn't complain, so I guess it's fine to leave it as is.

Reply:
You have a point here. Given that we know that `pydantic` will populate the value it's not correct to say it's an optional bool, it's a bool.
Will update it to make it clear.